java/ql/test/query-tests/security/CWE-532/Test.java

```diff
@@ -9,5 +9,7 @@ void test(String password, String authToken, String username, String nullToken,
         logger.error("Auth failed for: " + username); // Safe
         logger.error("Auth failed for: " + nullToken); // Safe
         logger.error("Auth failed for: " + stringTokenizer); // Safe
+        logger.error("Auth failed for: " + authToken.substring(4) + "..."); // Safe
+        logger.error("Auth failed for: " + authToken.substring(0,4) + "..."); // Safe
```

Comment:
I think it's worth testing that we don't just sanitize any substring operation.
```suggestion
        logger.error("Auth failed for: " + authToken.substring(0,4) + "..."); // Safe
        logger.error("Auth failed for: " + authToken.substring(0,8) + "..."); // $ Alert
```